### PR TITLE
split the result filed to metadata and inspect

### PR DESCRIPTION
**Pull request by chengjoey**
#### What type of this PR

Add one of the following kinds:
/kind bugfix

#### What this PR does / why we need it:
split the result filed to metadata and inspect, update metadata only when task callback

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda-org.erda.cloud/erda/dop/projects/387/issues/all?id=209158&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIl0sInN0YXRlcyI6WzQ0MDIsNzEwNCw3MTA1LDQ0MDMsNDQwNCw3MTA2LDQ0MDYsNDQwNyw0NDEyLDQ1MzgsNDQxMyw0NDE0LDQ0MTUsNDQxNl0sImFzc2lnbmVlSURzIjpbIjEwMDEyMDUiXX0%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=680&type=BUG)

#### Specified Reviewers:

/assign @your-reviewer

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by chengjoey** [comment]
image

convert to pipeline task dto:
image

File: .erda/migrations/pipeline/20211112-pipeline-task-metadata-inspect.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE `pipeline_tasks` ADD COLUMN `inspect` text COMMENT 'task的调度信息' AFTER `result`;
```
Comment (sfwn) [inline .erda/migrations/pipeline/20211112-pipeline-task-metadata-inspect.sql:1]: mediumtext

File: apistructs/pipeline_task.go [hunk 2]
```diff
@@ -62,6 +62,7 @@ type TaskContainer struct {
 	ContainerID string `json:"containerID"`
 }
 
+// PipelineTaskResult spec.pipeline task only use metadata, task dto has all fields
 type PipelineTaskResult struct {
 	Metadata    Metadata                   `json:"metadata,omitempty"`
 	Errors      []*PipelineTaskErrResponse `json:"errors,omitempty"`
```
Comment (sfwn) [inline apistructs/pipeline_task.go:68]: use embed struct

File: apistructs/pipeline_task.go [hunk 3]
```diff
@@ -195,7 +203,7 @@ func (o orderedResponses) Len() int           { return len(o) }
 func (o orderedResponses) Less(i, j int) bool { return o[i].Ctx.EndTime.Before(o[j].Ctx.EndTime) }
 func (o orderedResponses) Swap(i, j int)      { o[i], o[j] = o[j], o[i] }
 
-func (t *PipelineTaskResult) AppendError(newResponses ...*PipelineTaskErrResponse) []*PipelineTaskErrResponse {
+func (t *PipelineTaskInspect) AppendError(newResponses ...*PipelineTaskErrResponse) []*PipelineTaskErrResponse {
```
Comment (sfwn) [inline apistructs/pipeline_task.go:206]: use result is still ok

File: modules/pipeline/pipengine/actionexecutor/plugins/apitest/apitest.go [hunk 4]
```diff
@@ -96,7 +96,10 @@ func (d *define) Start(ctx context.Context, task *spec.PipelineTask) (interface{
 			return
 		}
 
-		meta := latestTask.Result.Metadata
+		var meta apistructs.Metadata
+		if latestTask.Result != nil {
+			meta = (*latestTask.Result).Metadata
```
Comment (sfwn) [inline modules/pipeline/pipengine/actionexecutor/plugins/apitest/apitest.go:101]: add a function for spec.task to get metadata.
judge `nil` each time is bad.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**